### PR TITLE
Fixing compilation error for generated Cordova projects on Android

### PR DESCRIPTION
**Pull request by bhariharan**

### CODE REVIEW

File: tools/postinstall-android.js [hunk 1]
```diff
@@ -74,6 +74,7 @@ replaceTextInFile(path.join(appProjectRoot, 'MobileSync', 'build.gradle'), oldSm
 replaceTextInFile(path.join(appProjectRoot, 'SalesforceHybrid', 'build.gradle'), oldMobileSyncDep, 'api project\(\':MobileSync\'\)');
 
 console.log('Fixing root level Gradle file for the generated app');
+replaceTextInFile(path.join(appProjectRoot, 'settings.gradle'), "include \":CordovaLib\"", "");
```
Comment (bhariharan) [inline tools/postinstall-android.js:77]: This was the problem - it has 2 references to `CordovaLib`, the one that is implicitly pulled in through `SalesforceHybrid` and the one coming from the generated project.

File: tools/postinstall-android.js [hunk 2]
```diff
@@ -95,24 +96,23 @@ if (data.indexOf("SalesforceHybrid") < 0)
     var oldAndroidDepTree = "android {";
     var newAndroidDepTree = "android {\n\tpackagingOptions {\n\t\texclude 'META-INF/LICENSE'\n\t\texclude 'META-INF/LICENSE.txt'\n\t\texclude 'META-INF/DEPENDENCIES'\n\t\texclude 'META-INF/NOTICE'\n\t}";
     replaceTextInFile(path.join(appProjectRoot, 'app', 'build.gradle'), oldAndroidDepTree, newAndroidDepTree);
-    var oldGradleToolsVersion = "com.android.tools.build:gradle:3.3.0";
-    var newGradleToolsVersion = "com.android.tools.build:gradle:3.5.1";
+    var oldGradleToolsVersion = "com.android.tools.build:gradle:4.0.0";
```
Comment (bhariharan) [inline tools/postinstall-android.js:99]: Updating these values to the correct values.

File: tools/postinstall-android.js [hunk 3]
```diff
@@ -95,24 +96,23 @@ if (data.indexOf("SalesforceHybrid") < 0)
...
+    var newGradleToolsVersion = "com.android.tools.build:gradle:4.2.0";
     replaceTextInFile(path.join(appProjectRoot, 'app', 'build.gradle'), oldGradleToolsVersion, newGradleToolsVersion);
-    replaceTextInFile(path.join(appProjectRoot, 'app', 'build.gradle'), 'mavenCentral()', 'google()');
```
Comment (bhariharan) [inline tools/postinstall-android.js:101]: We don't need this anymore. `Cordova` includes `google()` out of the box now.

File: tools/postinstall-android.js [hunk 4]
```diff
@@ -95,24 +96,23 @@ if (data.indexOf("SalesforceHybrid") < 0)
...
-var newGradleToolsVersion = "com.android.tools.build:gradle:4.0.0";
+var oldGradleToolsVersion = "com.android.tools.build:gradle:4.0.0";
+var newGradleToolsVersion = "com.android.tools.build:gradle:4.2.0";
 
 replaceTextInFile(path.join(appProjectRoot, 'build.gradle'), oldGradleToolsVersion, newGradleToolsVersion);
 replaceTextInFile(path.join(appProjectRoot, 'build.gradle'), /defaultBuildToolsVersion=.*/, '');
 replaceTextInFile(path.join(appProjectRoot, 'build.gradle'), /defaultMinSdkVersion=.*/, 'defaultMinSdkVersion=23');
-replaceTextInFile(path.join(appProjectRoot, 'build.gradle'), /defaultTargetSdkVersion=.*/, 'defaultTargetSdkVersion=29');
-replaceTextInFile(path.join(appProjectRoot, 'build.gradle'), /defaultCompileSdkVersion=.*/, 'defaultCompileSdkVersion=29');
+replaceTextInFile(path.join(appProjectRoot, 'build.gradle'), /defaultTargetSdkVersion=.*/, 'defaultTargetSdkVersion=30');
```
Comment (bhariharan) [inline tools/postinstall-android.js:114]: Should be `30`.

**Review by wmathurin — APPROVED** [review]

**Pull request merged**